setup.sh

```diff
@@ -1,10 +1,83 @@
 #!/usr/bin/env bash
 
-export SERVER_ADDRESS="$1"
-export HULY_SECRET="secret"
+# Define the default values
+HULY_VERSION="v0.6.295"
+DOCKER_NAME="huly"
+CONFIG_FILE="huly.conf"
 
-echo "Setting Huly Server Address: $SERVER_ADDRESS"
-envsubst < template.conf > nginx.conf
-envsubst < template.env > .env
 
-./use-version.sh v0.6.295
+if [ -f "$CONFIG_FILE" ]; then
+    source "$CONFIG_FILE"
+fi
+clear
+
+while true; do
+    read -p "Enter the host address (domain name or IP) [default: ${HOST_ADDRESS:-localhost}]: " input
+    _HOST_ADDRESS="${input:-${HOST_ADDRESS:-localhost}}"
+    #TODO: proper validation
+    #if [[ "$_HOST_ADDRESS" =~ ^[a-zA-Z0-9.-]+$ || "$_HOST_ADDRESS" =~ ^([0-9]{1,3}\.){3}[0-9]{1,3}$ ]]; then
+        break
+    #else
+    #   echo "Invalid host address. Please enter a valid domain name or IP."
+    #fi
+done
+clear
+
+while true; do
+    read -p "Enter the port for HTTP [default: ${HTTP_PORT:-80}]: " input
+    _HTTP_PORT="${input:-${HTTP_PORT:-80}}"
+    if [[ "$_HTTP_PORT" =~ ^[0-9]+$ && "$_HTTP_PORT" -ge 1 && "$_HTTP_PORT" -le 65535 ]]; then
+        break
+    else
+        echo "Invalid port. Please enter a number between 1 and 65535."
+    fi
+done
+clear
+# TODO: true/false
+while true; do
+    read -p "Will you serve Huly over SSL? (y/N) [default: ${SECURE}]: " SECURE_INPUT
```

Comment:
`${SECURE}` substs to either `s` or an empty string and that does not match the requested values (y/N).

Reply:
This one is still a WIP, I want to make it a true/false indicated by the TODO